bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/AndroidTVHandler.java

```diff
@@ -0,0 +1,261 @@
+        this.thingTypeUID = thingTypeUID;
+        this.thingID = this.getThing().getUID().getId();
+    }
+
+    public void setThingProperty(String property, String value) {
+        thing.setProperty(property, value);
+    }
+
+    public String getThingID() {
+        return this.thingID;
+    }
+
+    public void updateChannelState(String channel, State state) {
+        updateState(channel, state);
+    }
+
+    public ScheduledExecutorService getScheduler() {
+        return scheduler;
+    }
+
+    public void updateCDP(String channelName, Map<String, String> cdpMap) {
+        logger.trace("{} - Updating CDP for {}", this.thingID, channelName);
+        List<CommandOption> commandOptions = new ArrayList<CommandOption>();
+        cdpMap.forEach((key, value) -> commandOptions.add(new CommandOption(key, value)));
+        logger.trace("{} - CDP List: {}", this.thingID, commandOptions.toString());
+        commandDescriptionProvider.setCommandOptions(new ChannelUID(getThing().getUID(), channelName), commandOptions);
+    }
+
+    private void monitorThingStatus() {
+        synchronized (monitorThingStatusJobLock) {
+            checkThingStatus();
+            monitorThingStatusJob = scheduler.schedule(this::monitorThingStatus, THING_STATUS_FREQUENCY,
+                    TimeUnit.MILLISECONDS);
+        }
+    }
+
+    public void checkThingStatus() {
+        String statusMessage = "";
+        boolean failed = false;
+
+        GoogleTVConnectionManager googletvConnectionManager = this.googletvConnectionManager;
+        ShieldTVConnectionManager shieldtvConnectionManager = this.shieldtvConnectionManager;
+
+        if (googletvConnectionManager != null) {
+            if (!googletvConnectionManager.getLoggedIn()) {
+                statusMessage = "GoogleTV: " + googletvConnectionManager.getStatusMessage();
+                failed = true;
+            } else {
+                statusMessage = "GoogleTV: ONLINE";
+            }
+        }
+
+        if (THING_TYPE_SHIELDTV.equals(thingTypeUID)) {
+            if (shieldtvConnectionManager != null) {
+                if (!shieldtvConnectionManager.getLoggedIn()) {
+                    statusMessage = statusMessage + " | ShieldTV: " + shieldtvConnectionManager.getStatusMessage();
+                    failed = true;
+                } else {
+                    statusMessage = statusMessage + " | ShieldTV: ONLINE";
+                }
+            }
+        }
+
+        if (failed) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, statusMessage);
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Protocols Starting");
+
+        GoogleTVConfiguration googletvConfig = getConfigAs(GoogleTVConfiguration.class);
+        String ipAddress = googletvConfig.ipAddress;
+
+        if (ipAddress == null || ipAddress.isEmpty()) {
```

Comment:
See comment in `GoogleTVConfiguration`:
```suggestion
        if (ipAddress.isBlank()) {
```